### ISSUE TITLE
Can't generate scratch card for Registration

### ISSUE THREAD

**Issue by toliano** [ekumahost/my-complete-high-school-portal#1]
Am deeply in love with your project. But please, I can't generate scratch cards for Student to register on the platform

**Comment by ekumahost** [comment]
A field in your database table does not have default value.
 please enable error reporting in admin area. **/control/includes/configuration.php**
to see error message, then run this following query in your db
Next time you have issue give full details: "I can't generate scratch cards for Student to register" is not an issue, you must include error messages and screen shots if possible.

    ALTER TABLE `student_wallet_pins` CHANGE `used_by` `used_by` INT(8) NOT NULL DEFAULT '0';
also run 
     
    ALTER TABLE `reg_pins` CHANGE `used_by` `used_by` INT(8) NOT NULL DEFAULT '0';
     ALTER TABLE `reg_pins_old` CHANGE `used_by` `used_by` INT(8) NOT NULL DEFAULT '0';

**Comment by toliano** [comment]
Thanks a million. I duly followed your instruction in altering the database tables and have also activated the error reporting. 
1. When I tried to create the scratch cards I received the error message below
![scratch card](https://user-images.githubusercontent.com/38895569/66401908-d6101a80-e9db-11e9-9ebb-5cedd6887a03.PNG)


2. Then also when tying to switch the academic session back and front, I received the error message below:


![session](https://user-images.githubusercontent.com/38895569/66401945-e7f1bd80-e9db-11e9-9004-8090fa33eec5.PNG)
Thanks a million in advance. God bless you sir

**Comment by ekumahost** [comment]
global fix

     SET GLOBAL sql_mode='';

**Comment by toliano** [comment]
Thanks a lot sir. Then also another thing is when trying to register a
student from the front-end as *STUDENT ADMISSION, *I get this error message
about the captcha not correct below:
[image: CAPTCHA.PNG]

**Comment by ekumahost** [comment]
cant see the error message

**Comment by toliano** [comment]
Good morning sir and thanks a lot for yesterday. I have just little issue
left sir:

1. On the student registration page, after filling up all the fields and I
hit the SUBMIT BUTTON. It will always display CAPTCHA IS INCORRECT PLEASE
TRY AGAIN.
[image: CAPTCHA.PNG]

2. Am trying to change the SESSION but its still referring to me creating
the new year/session I want to change to of which I already did.

Thanks in advance sir

**Comment by ekumahost** [comment]
there is create session, if you are in 2018/2019 and you want to switch when you have not created 2019/2020 it should not work..

**Comment by ekumahost** [comment]
the catcha is working in my retest, what did you type in your captcha?

**Comment by toliano** [comment]
Hello sir, so sorry for the late reply. I have created the 2018/2019 and
even created more till 2023/2024.

And for the captcha. I always give the *SUM OF THE NUMBER GENERATED *in the
captcha field as a result. Thanks so much sir in advance.

**Comment by ekumahost** [comment]
The captcha is working fine, what could you be typing that your own dont
work. Go to the code and disable captcha check. Or launch teamviewer where
i can look at your issue.

**Comment by toliano** [comment]
Thanks so very much sir. I had to change some columns default value to NULL
in STUDENTBIO table and now the CAPTCHA is working on fine and good. I only
have just few things left to go great thanks for the help and instant
support you are providing to me. I can't just thanks you enough sir.

1. Firstly I would like to fix the switching of SESSION front & back. This
is how I created the new SESSION may be am wrong sir:
     a. Configuration
[image: sconfig.PNG]
     b. school year/session
[image: syear.PNG]
     c. I created about 4 more sessions
[image: session2.PNG]
     d. I clicked on class room/hall
[image: shall.PNG]
     e. I clicked on grade term days
[image: sgrade.PNG]
     f. Then I filled up the session I created from 1st Term to 3rd Term
with start and end date. Below are the screenshot sir
[image: session3.PNG]
g. switching
[image: sswitch.PNG]
h. error message
[image: serror.PNG]

THANKS IN ADVANCE SIR
